src/libstore/content-address.hh

```diff
@@ -39,12 +39,26 @@ enum struct FileIngestionMethod : uint8_t {
     /**
      * Flat-file hashing. Directly ingest the contents of a single file
      */
-    Flat = false,
+    Flat,
+
     /**
      * Recursive (or NAR) hashing. Serializes the file-system object in Nix
      * Archive format and ingest that
      */
-    Recursive = true
+    Recursive,
+
+    /**
+     * Git hashing. In particular files are hashed as git "blobs", and
+     * directories are hashed as git "trees".
```

Comment:
Do you intend to limit this to only git trees?

Reply:
Right now it does blobs and trees. I don't really care whether we support blobs or not.

Reply:
I think it would be less confusing. You don't need to start fetching data to know if you're dealing with a tree object or a blob object, and dealing with tree objects also seems more common than blob objects.

We are not able to express every store path anyways (executable files or symlinks at the root), so I'd rather have this simpler.